pysam/libcalignmentfile.pyx

```diff
@@ -550,16 +771,16 @@ cdef class AlignmentFile(HTSFile):
 
             # header structure (used for writing)
             if template:
-                self.header = bam_hdr_dup(template.header)
+                self.header = template.header.copy()
```

Comment:
Should check type of `template.header`.  Also, `copy` is not needed at the moment, since `AlignmentHeader`s are immutable.  May still be a good idea to keep the `copy` if/when they do change.